server/src-lib/Hasura/GraphQL/Transport/WebSocket/Server.hs

```diff
@@ -106,13 +107,18 @@ data WSServer a
   = WSServer
   { _wssLogger  :: L.Logger
   , _wssConnMap :: ConnMap a
+  , _wssShutdown :: STM.TChan ()
```

Comment:
Using a `TChan` here seems unnecessarily powerful. Why not just `_wssShouldShutdown :: TVar Bool`? You can efficiently block on that becoming `True` using `atomically (guard =<< readTVar shouldShutdownVar)`.

Reply:
Makes sense, I'll change that

Reply:
The result looks less elegant, as basically I'm replicating the internals of `readChan`, let me know if that is what you had in mind